src/isl/util.rs

```diff
@@ -134,6 +135,28 @@ impl PartialOrd for TimestampPrecision {
     }
 }
 
+impl Display for TimestampPrecision {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), std::fmt::Error> {
+        write!(
+            f,
+            "{}",
+            match &self {
+                TimestampPrecision::Year => "year",
+                TimestampPrecision::Month => "month",
+                TimestampPrecision::Day => "day",
+                TimestampPrecision::Minute => "minute",
+                TimestampPrecision::Second => "second",
+                TimestampPrecision::Millisecond => "millisecond",
+                TimestampPrecision::Microsecond => "microsecond",
+                TimestampPrecision::Nanosecond => "nanosecond",
+                TimestampPrecision::OtherFractionalSeconds(_) => {
+                    "fractional second"
+                }
```

Comment:
question—any particular reason that we are leaving out the `scale` of `OtherFractionalSeconds` instead of  displaying something like `format!("fractional seconds (10e{})", scale*-1)`?

Reply:
No particular reason. Thought to simplify by just saying fractional seconds but I can see that the scale information will be useful here.